dlk/python/dlk/core/optimizer.py

```diff
@@ -207,6 +227,47 @@ def pass_propagate_quantization_details_into_conv(graph: Graph) -> None:
             # TODO: check if the quantizers use same n_bits
 
 
+def pass_precompute_batchnormalization(graph: Graph) -> None:
+    """TODO: write datails.
+
+    Parameters
+    ----------
+    graph : Graph
+        The input graph. It will be modified in-place.
+    """
+    exec_list = [n for n in sort_graph(graph) if n.op_type == 'BatchNormalization']
+
+    for m in exec_list:
+        bn_gamma = m.input_ops['scale'].data
+        bn_beta = m.input_ops['B'].data
+        bn_mean = m.input_ops['mean'].data
+        bn_variance = m.input_ops['var'].data
+        bn_epsilon = m.epsilon
+
+        bn_scale = calc_scale(bn_gamma, bn_variance, bn_epsilon)
+        bn_shift = calc_shift(bn_scale, bn_beta, bn_mean)
+
+        bn_shift_constant = Constant(
+            m.name + '_new_shift',
+            Float32,
+            data=bn_shift
+        )
+        bn_scale_constant = Constant(
+            m.name + '_new_scale',
+            Float32,
+            data=bn_scale
+        )
+
+        graph.add_op(bn_shift_constant)
+        graph.add_op(bn_scale_constant)
+
+        bn_shift_constant.add_output('output', m)
+        bn_scale_constant.add_output('output', m)
+
+        m.add_input('prep_scale', bn_scale_constant)
+        m.add_input('shift', bn_shift_constant)
```

Comment:
I think it is better to add `Add` and `Mul` operator instead of `BatchNorm` 🤔🤔🤔 
Why do you choice to use `BatchNorm` operator adding `scale` and `shift` ?

Reply:
At least, current implementation is better than `Mul` and `Add` in terms of memory access.
In addition, it is better to have fuse with Convolution when possible.